### ISSUE TITLE
New Rule: `UnnecessaryRequireNotNull`

### ISSUE THREAD

**Issue by pbreault** [detekt/detekt#5204]
## Expected Behavior of the rule
Would there be any appetite for a new `UnnecessaryRequireNotNull` rule?

This would be similar to [UnnecessaryNotNullOperator](https://detekt.dev/docs/rules/potential-bugs/#unnecessarynotnulloperator) but it would flag calls to [requireNotNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require-not-null.html)  with a non-nullable parameter.

```kotlin
// non compliant code 
var string = "foo"
println(requireNotNull(string))

// compliant code
var string : String? = "foo"
println(requireNotNull(string))
```

## Context
We have seen optional field become required after refactorings, leaving behind unnecessary calls to `requireNotNull`. 
We have built a custom rule to detect these cases and we would like to contribute it back.

**Comment by 3flex** [comment]
I think this would be a great addition, and should also check for the same case with `checkNotNull`

**Comment by schalkms** [comment]
I second this. This rule would be a great addition to detekt.
Shall the rule be called `UnnecessaryNotNullCheck`?

@pbreault may I ask you to take a stab at this rule?

**Comment by pbreault** [comment]
I think a single `UnnecessaryNotNullCheck ` rule that encompasses checkNotNull and requireNotNull makes sense. Definitely more future-proof. I'll take a stab at it.
